drivers/espi/espi_emul.c

```diff
@@ -19,170 +19,480 @@ LOG_MODULE_REGISTER(espi_emul_ctlr);
 #include <zephyr/drivers/emul.h>
 #include <zephyr/drivers/espi.h>
 #include <zephyr/drivers/espi_emul.h>
-#include "espi_utils.h"
+#include <zephyr/drivers/espi/espi_utils.h>
+
+/** Declare the default state of virtual wires */
+const static struct espi_emul_vw_data vw_state_default[] = {
+	{
+		.sig = ESPI_VWIRE_SIGNAL_OOB_RST_ACK,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_WAKE,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_PME,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_ERR_FATAL,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_ERR_NON_FATAL,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_SLV_BOOT_STS,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_SCI,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_SMI,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_RST_CPU_INIT,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_HOST_RST_ACK,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_SUS_ACK,
+		.level = 0,
+	},
+	{
+		.sig = ESPI_VWIRE_SIGNAL_DNX_ACK,
+		.level = 0,
+	},
+};
+
+#define NUMBER_OF_VWIRES ARRAY_SIZE(vw_state_default)
 
 /** Working data for the controller */
 struct espi_emul_data {
-	/* List of struct espi_emul associated with the device */
-	sys_slist_t emuls;
+	/* eSPI emul associated with the device */
+	struct espi_emul *emul;
```

Comment:
This doesn't look right to me, it needs to be a list. You can have multiple peripherals attached to the same bus.

Reply:
Not exactly from our side. The host can have multiple peripherals, but our EC is peripheral-only.
eSPI API doesn't contain any chip-selecting, we just send data to the bus, not to the peripheral, and the host is signaled using interrupts that it needs to read our data.
The best example is the keyboard controller which can be embedded in EC. Using our API, we send some requests to the KBC but using the bus connected to the host (bus is our destination device). Theoretically, we are sending these data to host, but the chip drivers intercepts these requests and handle them internally (so do this emulator). We could have another eSPI bus only for KBC, but then, the KBC would need to have peripheral bus to controller EC and another peripheral bus to controller host (because host needs to read the data from KBC using the I/O through eSPI/LPC).
Looking at this, from EC (zephyr) side, we can't be connected to more than one (host) peripheral on the bus.

Reply:
Maybe we just need to make it more explicit that this is only a host emulator, not a general purpose espi emulator then.

Reply:
I tried to change the descriptions and Kconfigs to be more accurate. Let me know if there is more I should do with it.